### ISSUE TITLE
MPC use case

### ISSUE THREAD

**Issue by jbcaillau** [control-toolbox/OptimalControl.jl#250]
@agustinyabo should be rather easy to test MPC (fixed / moving horizon) implementing a basic loop (see, *e.g.*, this [simple example](https://github.com/pns-mam/commande/blob/master/tp3/navigation-mpc-sol.ipynb)) on the nice example you treated in the ref. below:

Yabo, A.; Caillau, J.-B., Gouzé, J.-L.
[Hierarchical MPC applied to bacterial resource allocation and metabolite synthesis](http://caillau.perso.math.cnrs.fr/research/cdc-2021b.pdf)
*IEEE Conference on Decision and Control* (2021), 667-672
Proceedings of the 60th CDC, Austin, December 2021

**Comment by ocots** [comment]
@jbcaillau We need a [template repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) to facilitate adding applications.

**Comment by ocots** [comment]
@jbcaillau Do we add in "Tutorials and Advanced Features" a simple use of MPC?

**Comment by jbcaillau** [comment]
yes, would en nice. we could reuse sth like this (Zermelo, JuMP version):
https://github.com/pns-mam/commande/blob/main/tp3/tp3-sol.md

**Comment by ocots** [comment]
I have converted your example from JuMP to OptimalControl: https://gitlab.irit.fr/toc/etu-n7/controle-optimal/-/blob/master/tp/mpc-navigation.ipynb.

**Comment by jbcaillau** [comment]
nice. ajouter ensuite la sol. (in english) dans les tutos ?

> I have converted your example from JuMP to OptimalControl: https://gitlab.irit.fr/toc/etu-n7/controle-optimal/-/blob/master/tp/mpc-navigation.ipynb.

**Comment by ocots** [comment]
[mpc-navigation-prof.pdf](https://github.com/user-attachments/files/20274143/mpc-navigation-prof.pdf)

**Comment by ocots** [comment]
Oui j'ajouterai un tuto qui reprend ça. Par contre, il y a des limites. Là, j'utilise `disc_method=:euler` car avec les trapèzes, le contrôle il oscille trop, et avec un Gauss-Legendre, on converge vers une autre solution locale. Après, ce n'est pas très robuste, on peut facilement ne pas converger.

**Comment by ocots** [comment]
We can add this application:

Yabo, A.; Caillau, J.-B., Gouzé, J.-L.
[Hierarchical MPC applied to bacterial resource allocation and metabolite synthesis](http://caillau.perso.math.cnrs.fr/research/cdc-2021b.pdf)
IEEE Conference on Decision and Control (2021), 667-672
Proceedings of the 60th CDC, Austin, December 2021

@jbcaillau A new tutorial or a new application or inside https://agustinyabo.github.io/PWLdynamics.jl.

**Comment by jbcaillau** [comment]
@agustinyabo ?

> We can add this application:
> 
> Yabo, A.; Caillau, J.-B., Gouzé, J.-L. [Hierarchical MPC applied to bacterial resource allocation and metabolite synthesis](http://caillau.perso.math.cnrs.fr/research/cdc-2021b.pdf) IEEE Conference on Decision and Control (2021), 667-672 Proceedings of the 60th CDC, Austin, December 2021
> 
> [@jbcaillau](https://github.com/jbcaillau) A new tutorial or a new application or inside https://agustinyabo.github.io/PWLdynamics.jl.

**Comment by agustinyabo** [comment]
It's a good idea, I'll think about it. For some time, I wanted to try some robust MPC methods, like the classical Min-Max MPC from:

Campo, P. J., & Morari, M. (1987, June). Robust model predictive control. In 1987 American control conference. IEEE.

I don't know if this Hierarchical MPC is the best example, maybe some other problem with reference tracking?
